visual-java/src/main/java/com/saucelabs/visual/VisualApi.java

```diff
@@ -659,11 +659,20 @@ private void sauceVisualCheckLocal(String snapshotName, CheckOptions options) {
       }
     }
 
+    List<RegionIn> visibleIgnoreRegions = new ArrayList<>();
     for (RegionIn region : ignoreRegions) {
-      Point newPoint =
-          CartesianHelpers.relativeTo(viewport.getPoint(), new Point(region.getX(), region.getY()));
-      region.setX(newPoint.x);
-      region.setY(newPoint.y);
+      Rectangle regionRect =
+          new Rectangle(region.getX(), region.getY(), region.getHeight(), region.getWidth());
```

Comment:
Shouldn't this account for the relative changes first ('before' of this diff) prior to checking if they're in the viewport?

Pseudocode: 

```
// mutate point based on viewport changes

// If intersects with viewport, then add as a visible region
```

Reply:
No, as `viewport` and `regionRect` are absolute here (relative to the `(0, 0)` point).

We modify the points to be relative to the image itself (which is relative to `(viewport.x, viewport.y)`) afterwards.